### DISCUSSION TITLE
Client IP authentication behind a proxy

### DISCUSSION THREAD

**Discussion by boamaod** [abersheeran/asgi-ratelimit#55] [Q&A] [answered]
In a commit https://github.com/abersheeran/asgi-ratelimit/commit/0867599f79e89008025ec552160be79cfe1d73f0 checking `X-Forwarded-For` as part of client IP authentication was removed and after that also README [states that](https://github.com/abersheeran/asgi-ratelimit/blob/9af3a7a32e9eab58460ed67c76959ed13bdfc57c/README.md?plain=1#L188):

> Note: this auth method will not work if your IP address (such as 127.0.0.1 etc) is not allocated for public networks.

Do I understand correctly that this is mostly for simplifying and unifying the authentication code and does not mean to imply that running behind a proxy excludes client IP authentication? Or is there a reason for a proxy not to forward `X-Real-Ip` header to make client IP authentication work?

If not, maybe also the README should suggest configuring forwarded headers to make `X-Real-Ip` directly accessible or handle authentication in custom function, because otherwise it gives an impression that rate limiting behind proxy is not meant to work because of some inherent reason that makes it impossible. Or am I missing the point here?

**Comment by abersheeran** [comment, marked as answer]
X-Forwarded-For is easy to be forged, and the implementation of X-Real-IP in each proxy server is more secure. So the built-in Client IP function removes the judgment of X-Forwarded-For.

**Reply by abersheeran** [reply]
By the way, you are welcome to initiate a pull request to improve the documentation together.
